apps/olat-api/src/services.ts

```diff
@@ -24,19 +24,20 @@ export async function getCourses(provider: string, providerAccountId: string) {
     select: {
       user: {
         select: {
-          courses: {
-            where: { isArchived: false },
-            select: { id: true, name: true },
+          // find all shared non-archived courses (permission level irrelevant - read permissions are sufficient)
+          objects: {
+            where: { courseId: { not: null }, course: { isArchived: false } },
+            select: { course: { select: { id: true, name: true } } },
```

Comment:
@jabbadizzleCode Through the "objects" on the user we can directly join all objects of a type (going through the derivedPermissions table). With the filters added here, we only include courses that have the archived boolean set to false